From 8ca40a6454f053a262926794b96024bd02e0cdc5 Mon Sep 17 00:00:00 2001
From: Milian Wolff <mail@milianw.de>
Date: Wed, 31 May 2023 09:31:02 +0200
Subject: [PATCH] Fix compile with newer astyle

astyle::streamoff was always just an alias for std::streamoff,
see also the upstream commit [1] that removed the
`using namespace std` in the `astyle` namespace which breaks this
code of our here now.

[1]: https://gitlab.com/saalen/astyle/-/commit/0add646b21ee329b231ef7c5ab6963a94fb8935b
---
 plugins/astyle/astyle_stringiterator.cpp | 4 ++--
 plugins/astyle/astyle_stringiterator.h   | 5 +++--
 2 files changed, 5 insertions(+), 4 deletions(-)

diff --git a/plugins/astyle/astyle_stringiterator.cpp b/plugins/astyle/astyle_stringiterator.cpp
index 02f8fb4264..a611869581 100644
--- a/plugins/astyle/astyle_stringiterator.cpp
+++ b/plugins/astyle/astyle_stringiterator.cpp
@@ -22,7 +22,7 @@ AStyleStringIterator::~AStyleStringIterator()
 {
 }
 
-astyle::streamoff AStyleStringIterator::tellg()
+std::streamoff AStyleStringIterator::tellg()
 {
   return m_is.pos();
 }
@@ -59,7 +59,7 @@ void AStyleStringIterator::peekReset()
     m_peekStart = -1; // invalid
 }
 
-astyle::streamoff AStyleStringIterator::getPeekStart() const
+std::streamoff AStyleStringIterator::getPeekStart() const
 {
     // NOTE: we're not entirely sure if this is the correct implementation.
     // we're trying to work-around https://bugs.kde.org/show_bug.cgi?id=399048
diff --git a/plugins/astyle/astyle_stringiterator.h b/plugins/astyle/astyle_stringiterator.h
index f63a59c611..713800c0f3 100644
--- a/plugins/astyle/astyle_stringiterator.h
+++ b/plugins/astyle/astyle_stringiterator.h
@@ -21,13 +21,14 @@ public:
     explicit AStyleStringIterator(const QString &string);
     ~AStyleStringIterator() override;
 
-    astyle::streamoff tellg() override;
+    std::streamoff tellg() override;
     int getStreamLength() const override;
     bool hasMoreLines() const override;
     std::string nextLine(bool emptyLineWasDeleted = false) override;
     std::string peekNextLine() override;
     void peekReset() override;
-    astyle::streamoff getPeekStart() const override;
+    std::streamoff getPeekStart() const override;
+
 private:
     QString m_content;
     QTextStream m_is;
-- 
GitLab

